packages/interactive-messages/src/http-handler.ts

```diff
@@ -55,26 +53,28 @@ export function createHTTPHandler(adapter) {
   /**
    * Method to verify signature of requests
    *
-   * @param {string} signingSecret - Signing secret used to verify request signature
-   * @param {Object} requestHeaders - Request headers
-   * @param {string} body - Raw body string
-   * @returns {boolean} Indicates if request is verified
+   * @param signingSecret - Signing secret used to verify request signature
+   * @param requestHeaders - Request headers
+   * @param body - Raw body string
+   * @returns Indicates if request is verified
    */
-  function verifyRequestSignature(signingSecret, requestHeaders, body) {
+  function verifyRequestSignature(
+    signingSecret: string,
+    requestHeaders: Record<string, string>,
```

Comment:
Should we instead use the `IncomingHttpHeaders` type from `@types/node`?

Reply:
We *could*, and that'd be very correct! But then we'd need to either handle that headers can technically be arrays or add `as string | undefined` in a lot of places.

As an aside, I didn't like this signature because the proper version of this would be `Record<string, string | undefined>` but there is no handling in place for `undefined` values.

Reply:
Your concerns with using `IncomingHttpHeaders` are very keen! Can we offer a more useful type than `Record<string, string>`? We can create an interface that has properties `x-slack-signature: string;` and `x-slack-request-timestamp: number;`. In fact, that interface can extend `IncomingHttpHeaders` (where it will get an index signature), and the comment block can add helpful comment about being able to use the `req.headers` where `req` is an incoming request.

Reply:
😕 But then users would need to pass `req.headers as VerificationHeaders` to assert that those headers "will" be present.

(Also, it'd be `'x-slack-request-timestamp': string`)

Reply:
At face value, it does seem like it would be unergonomic. But in my opinion, the type assertion is a **good thing** for people's code. It forces a user to understand that using `verifyRequestSignature()` requires them to think about whether those specific headers will be present as they write the call site.

For the majority of the users of this package, they will not use `verifyRequestSignature()` directly. It should only "get in the way" for advanced users, who I assume would appreciate a little extra guidance.

(nice catch!)